tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -142,17 +146,93 @@ private static string ConstructErrorMessage(List<IError> executionErrors)
                     if (message != null)
                     {
                         builder.AppendLine($"{tab + tab}\"message\": \"{message.Replace("\r", "\\r").Replace("\n", "\\n")}\",");
+                        eventAttributes.Add(new KeyValuePair<string, object>("message", message));
+                    }
+
+                    var locations = executionError.Locations;
+                    if (locations != null)
+                    {
+                        ConstructErrorLocationsMessage(builder, tab, locations);
+
+                        var joinedLocations = new List<string>();
+                        foreach (var location in (System.Collections.IEnumerable)locations)
+                        {
+                            if (location.TryDuckCast<ErrorLocationStruct>(out var locationProxy))
+                            {
+                                joinedLocations.Add($"{locationProxy.Line}:{locationProxy.Column}");
+                            }
+                        }
+
+                        eventAttributes.Add(new KeyValuePair<string, object>("locations", joinedLocations.ToArray()));
                     }
 
-                    ConstructErrorLocationsMessage(builder, tab, executionError.Locations);
                     builder.AppendLine($"{tab}}},");
+
+                    var path = executionError.Path;
+                    if (path.Name != null)
+                    {
+                        var pathAttribute = new List<string>();
+                        pathAttribute.Add(path.Name);
+                        eventAttributes.Add(new KeyValuePair<string, object>("path", pathAttribute.ToArray()));
+                    }
+
+                    var code = executionError.Code;
+                    if (code != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("code", code));
+                    }
+
+                    var exception = executionError.Exception;
+                    if (exception != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("stacktrace", exception.StackTrace));
+                    }
+
+                    var extensions = executionError.Extensions;
+                    if (extensions != null)
+                    {
+                        var configuredExtensions = Tracer.Instance.Settings.GraphQLErrorExtensions;
+
+                        var keys = extensions.Keys.ToList();
```

Comment:
nit: This `ToList()` call is going to allocate a new `List<string>`. Instead, I suggest removing this line and do your iterating with `foreach (KeyValuePair<string, object> kvp in extensions)`